### PR TITLE
Reduced airdrop

### PR DESCRIPTION
**Pull request by setavenger**
Introduces the feature to airdrop keys (Xpubs) during the MultiSig key creation phase. A lot less complexity than the [old PR](https://github.com/BlueWallet/BlueWallet/pull/5660).

### CODE REVIEW

File: class/deeplink-schema-match.js [hunk 1]
```diff
@@ -377,6 +390,20 @@ class DeeplinkSchemaMatch {
     return text.startsWith('widget?action=');
   }
 
+  static hasValidKeysForMultiSigZpub(str) {
```
Comment (Overtorment) [inline class/deeplink-schema-match.js:393]: needs a unit test

File: ios/BlueWallet/Info.plist [hunk 2]
```diff
@@ -251,6 +263,29 @@
...
+				<key>public.filename-extension</key>
+				<array>
+					<string>json</string>
+				</array>
+				<key>public.mime-type</key>
+				<array>
+					<string>application/json</string>
+				</array>
+			</dict>
+		</dict>
```
Comment (Overtorment) [inline ios/BlueWallet/Info.plist:288]: @marcosrdz is this legit?

File: loc/en.json [hunk 3]
```diff
@@ -482,6 +482,8 @@
     "header": "Send",
     "share": "Share",
     "view": "View",
+    "shared_key_detected": "Shared cosigner",
+    "shared_key_detected_question": "A cosigner was shared with you, do you want to import it?",
```
Comment (Overtorment) [inline loc/en.json:486]: @ncoelho what would be the best wording?
less technical

File: screen/wallets/addMultisigStep2.js [hunk 4]
```diff
@@ -75,6 +77,35 @@ const WalletsAddMultisigStep2 = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  useEffect(() => {
+    console.log(currentSharedCosigner);
+    if (currentSharedCosigner) {
+      Alert.alert(
```
Comment (Overtorment) [inline screen/wallets/addMultisigStep2.js:83]: we have a `confirm` helper which does exactly this.

see `helpers/confirm.ts`


image

**Review by Overtorment — CHANGES_REQUESTED** [review]
mostly LGTM, mergeable, a few minor issues.
diff is considerably smaller, and easier to understand! me gusto.

im even starting to have doubts it will even work lol

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: screen/wallets/addMultisigStep2.js [hunk 5]
```diff
@@ -253,25 +269,31 @@ const WalletsAddMultisigStep2 = () => {
...
-    let fp;
-    try {
-      fp = await prompt(loc.multisig.input_fp, loc.multisig.input_fp_explain, true, 'plain-text');
-      fp = (fp + '').toUpperCase();
-      if (!MultisigHDWallet.isFpValid(fp)) fp = '00000000';
-    } catch {
-      return setIsLoading(false);
+    if (fp) {
+      //  do nothing, it's already set
+    } else {
```
Comment (limpbrains) [inline screen/wallets/addMultisigStep2.js:274]: That is an interesting way of writing `if (!fp)`

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: ios/BlueWallet/Info.plist (hunk 2 above)
Comment (marcosrdz) [reply ios/BlueWallet/Info.plist:288]: Yup

**Review by marcosrdz — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Review by Overtorment — APPROVED** [review]

**Pull request merged**